Service development

SpicedPine opened the pull request:
К конкретным коммитам больше информации написал и кое-какие вопросы, но в целом с этого пр-а может кровь из глаз потечь...

Commit:
Добавил DAO интерфейсы и их реализацию, отвечают за crud операции.

Commit:
Добавил добавил getter/setter

Commit:
Добавил классы для парсинга formedPrescription в PrescriptionScratch.

Только вот я не уверен, вдруг осознал, что по дефолту List в java immutable, поэтому по-идее методы для парсинга не будут работать, для того, чтобы они работали, получается нужно закидывать конкретную реализацию типа ArrayList в параметрах вызываемых методов?

Ну и вообще, опять ощущения, что я как-то грязно работаю с list-ом, да и сигнатура метода странная, в этом плане все ли правильно или нет? (анализатор текста тоже на ворнинги ругается, поэтому, скорее всего делаю что-то не то)

+ работа с исключениями, думаю там, где добавил в сигнатуру просто кидание Exception стоит создать свой Exception конкретный и его пробрасывать?

Commit:
Просто добавил enum для дней недели и перенес остальные в другой пакет

Commit:
Начал писать и понял, что не совсем понимаю одну вещь, подробнее в тг напишу.

damintsew commented on service/src/main/java/com/noskov/school/service/imp/prescription/PrescriptionBuilder.java:
```diff
@@ -0,0 +1,71 @@
+    public PrescriptionScratch parsePrescription(String prescription) throws Exception {
+        PrescriptionScratch scratch = new PrescriptionScratch();
+        List<String> list = Arrays.asList(prescription.split(" "));
+        if (list.get(0).equalsIgnoreCase(TherapyType.PROCEDURE.toString())){
```
Давай вот тут просто передавать поток управления в класс `ProcedureParser.parseProcedure`

Верхнеуровне это будет выглядеть примерно так:

TherapyType therapyType = extractType(prescription)
if (therapyType  == TherapyType.PROCEDURE) {
   PrescriptionScratch  parsedResult =  ProcedureParser.parseProcedure(prescription)
} else if( .... TherapyType.MEDICINE) {
  PrescriptionScratch  parsedResult =MedicineParser.parseMedicine(prescription);
}

damintsew replied:
И дальше мы спускаемся в в каждый из классов MedicineParser / ProcedureParser и там полностью формируется PrescriptionScratch

Commit:
Попробовал реализовать так, как ты говорил.
Вот только мне кажется, что тут отдельные классы, такие как QuantityParser или TimePeriodParser лишние и их методы можно раскидать прямиком в PeriodParser и TimePatternParser, думаю так более читабельный код будет, нет?

Ну и ещё тут анализатор немного ругается, что catch блоки пустые, так вообще можно делать или не стоит?

Commit:
Поправил синтаксис в ci

Commit:
Добавил методы для доставания внутренних полей объектов вложенных классов, не уверен, что это идеологическип правильно, но как доставать иначе поля не в том же пакете находясь не совсем понятно...

Commit:
Т.к. добавил в PrescriptionScratch.Type новое поле (название лекарства/процедуры), то поменял методы для парсинга.

P.s.пока имел ввиду, что название состоит из одного слова.

Commit:
Попробовал добавить имплементацию для генерации эвентов, но походу написания пришло в голову, что делаю что-то не так и всё как-то слишком грязно, поэтому вопрос: вообще в принципе идея того, как имплеменчу норм или стоит совсем с другой какой-то стороны заходить?

Commit:
исправил имя локальной переменной

Commit:
Добавил поле для количества дней в периоде.

Commit:
Немного поменял метод генерации эвентов,учтя пару других случаев, если сама идея такой реализации норм, то естественно разобью этот огромный метод на более маленькие + вынесу дублирующийся код в отдельные методы

damintsew commented on service/src/main/java/com/noskov/school/service/imp/EventGenerationServiceImp.java:
```diff
@@ -60,35 +56,85 @@ public void generateEvents(PrescriptionPO prescription) {
-                        EventPO event = new EventPO(patient,LocalDateTime.of(weekDate,time),EventStatus.PLANNED,therapyType);
-                        generatedEvents.add(event);
-                        });
+                if (weekDaysSet.isEmpty()){
```
ctrl + alt + M - extract method
Выдели в отдельный метод и дай ему название, которое описывает его

damintsew commented on service/src/main/java/com/noskov/school/service/imp/EventGenerationServiceImp.java:
```diff
@@ -60,35 +56,85 @@ public void generateEvents(PrescriptionPO prescription) {
+                        }
+                    }
+                } else {
+                    for (String dayOfWeek : weekDaysSet) {
```
ctrl + alt + M - extract method

damintsew commented on service/src/main/java/com/noskov/school/service/imp/EventGenerationServiceImp.java:
```diff
@@ -60,35 +56,85 @@ public void generateEvents(PrescriptionPO prescription) {
-                date = date.plus(1, ChronoUnit.valueOf(periodTimePeriod.toString().toUpperCase()+"s"));
             }
-        } else if(timePatternTimePeriod == TimePeriods.DAY){
+        } else if(timePatternTimePeriod == TimePeriods.DAY) {
```
ctrl + alt + M - extract method

damintsew commented on service/src/main/java/com/noskov/school/service/imp/EventGenerationServiceImp.java:
```diff
@@ -60,35 +56,85 @@ public void generateEvents(PrescriptionPO prescription) {
+                }
+                date = date.plus(1, ChronoUnit.valueOf(periodTimePeriod.toString().toUpperCase() + "s"));
             }
         } else if(timePatternTimePeriod == TimePeriods.Month){
```
ctrl + alt + M - extract method

Commit:
Причесал метод по твоим комментам.
